### ISSUE TITLE
Build command sometimes has to wait 5-10 seconds on "Fetching Build Tasks..."

### ISSUE THREAD

**Issue by sborkowski** [microsoft/vscode-cpptools#10474]
### Environment

- OS and Version: Windows 11 Pro 10.0.22621
- VS Code Version: 1.75.0
- C/C++ Extension Version: 1.14.1
- If using SSH remote, specify OS of remote machine: N/A

### Bug Summary and Steps to Reproduce

Bug Summary:

Issuing the build command (Ctrl+B) may sometimes cause VSCode to spend 5-10 seconds "Fetching Build Tasks..." (in status bar) before showing the build tasks menu. I've narrowed down the culprit to the CppTools extension and its C/C++ task provider that suggests a task depending on the file type, aka "detected tasks" in the build menu. I was able to fix this issue by changing the "Task: Auto Detect" setting to "off".

Steps to reproduce:
1. Open a workspace/folder with a C/C++ file.
2. Press Ctrl+B and wait for the menu to show.
3. If menu shows immediately, press escape.
4. Repeat steps 2-3 until you see "Fetching Build Tasks..." in the status bar.

Expected behavior:
There shouldn't be any wait for the build tasks menu to show, especially after its already been shown once before. It should be instantaneous.

**Comment by elahehrashedi** [comment]
Is this a CMake Project? CMake Task Provider will only provide tasks if the project is a CMake Project (which then makes the CMake Tools extension activated and then the cmake task provider will try to provide a list of tasks). There are also other extensions that may provide tasks too. So we may need more info about the project to see which task providers in different extensions are being activated.
for example, I have a CMake Project, in which both C/C++ and CMake tasks providers will provide tasks when running "Task: Run Build Task":
![image](https://user-images.githubusercontent.com/16143955/217612099-1b89142b-a4c9-4aba-849b-daa110be749d.png)

**Comment by sborkowski** [comment]
No, it's not a CMake project, and the only task that is suggested is the "C/C++: cl.exe build active file" one. I also narrowed it down to the cpptools extension by disabling/removing all others to see if the issue remained, and it did. I also tried rolling back several versions of vscode and the cpptools extension, but no luck. I will note that it wasn't until I upgraded to Win 11 that I started noticing this issue. Personally, I don't use the auto detected tasks, so I'm fine with disabling the feature altogether. I just figured I'd mention it as it seemed strange to me that it would randomly take on the order of 5-10 seconds to show a menu that had just been shown, and it took no time at all.

**Comment by sean-mcmanus** [comment]
Our team isn't able to repro this issue. Are you able to identify a repro requirement or are you able to identify what is going on when it says "Fetching Build Tasks..."? It's possible you might be able to find the cause by debugging the extension from the TypeScript: https://github.com/microsoft/vscode-cpptools/blob/main/Documentation/Building%20the%20Extension.md . Although be aware that in the next couple days we plan to update the Release branch for 1.14.3 so after that happens you'd need the 1.14.2-3 cpptools/-srv binaries.
